dbms/src/Flash/Pipeline/Schedule/TaskQueues/FIFOTaskQueue.cpp

```diff
@@ -12,23 +12,29 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <Flash/Pipeline/Schedule/TaskQueues/FiFOTaskQueue.h>
+#include <Flash/Pipeline/Schedule/TaskQueues/FIFOTaskQueue.h>
+#include <Flash/Pipeline/Schedule/Tasks/TaskHelper.h>
 #include <assert.h>
 #include <common/likely.h>
 
 namespace DB
 {
-bool FIFOTaskQueue::take(TaskPtr & task) noexcept
+FIFOTaskQueue::~FIFOTaskQueue()
+{
+    RUNTIME_ASSERT(task_queue.empty(), "all task should be taken before it is destructed");
```

Comment:
```suggestion
    RUNTIME_CHECK(task_queue.empty(), "all task should be taken before it is destructed");
```

Reply:
Because the destructor function in C++11 is noexcept, throwing an error can also lead to crashes and `Terminate called without an active exception`, so here we use RUNTIME_ASSERT.